sklearn/pipeline.py

```diff
@@ -792,6 +792,7 @@ class FeatureUnion(TransformerMixin, _BaseComposition):
     transformer_weights : dict, default=None
         Multiplicative weights for features per transformer.
         Keys are transformer names, values the weights.
+        Raises KeyError if key not present in ``transformer_list``.
```

Comment:
Applying the suggestion of @thomasjpfan you will need to change the error type as well.

```suggestion
        Raises ValueError if key not present in ``transformer_list``.
```